database.js

```diff
@@ -0,0 +1,39 @@
+// Load app configuration
+const config = require('config');
+// Load mongoose
+const Mongoose = require('mongoose');
+
+const connectDatabase = (callback) => {
+	// Calling Mongoose connect method with database URI will create a connection.
+	Mongoose.connect(config.get('app.dbConfig.url'));
+
+	const database = Mongoose.connection;
+	// If connection successfully established
+	database.once('connected', () => {
+		console.log('Connection with database succeeded.');
+		// returning status as true on success
+		callback(true);
+	});
+
+	// If connection throws an error
+	database.on('error', (error) => {
+		console.log('Connection with database failed.');
+		console.log(error);
+		// returning status as false on failure
+		callback(false);
```

Comment:
Instead of returning `true` or `false`, use `error` as the first argument. Then call with no argument if the connection is successful else call with `error` if failed. In the file where it is used, just check if `if (error)` and proceed accordingly. You can also remove `console.log` from here and added it the `index.js` file where it is used.